src/Akka.Persistence.Linq2Db.Data.Compatibility.Tests/DataCompatibilitySpec.cs

```diff
@@ -0,0 +1,133 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="DataCompatibilitySpec.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Hosting;
+using Akka.Persistence.Linq2Db.Data.Compatibility.Tests.Internal;
+using Akka.Persistence.Query;
+using Akka.Persistence.Sql.Compat.Common;
+using Akka.Persistence.Sql.Linq2Db.Query;
+using Akka.Streams;
+using Akka.Streams.Dsl;
+using Xunit;
+using Xunit.Abstractions;
+using FluentAssertions;
+
+namespace Akka.Persistence.Linq2Db.Data.Compatibility.Tests
+{
+    // Default compatibility spec, logical delete turned off
+    public abstract class DataCompatibilitySpec<T> : DataCompatibilitySpecBase<T> where T: ITestContainer, new()
+    {
+        protected DataCompatibilitySpec(ITestOutputHelper output): base(output)
+        {
+        }
+
+        protected sealed override void Setup(AkkaConfigurationBuilder builder, IServiceProvider provider)
+        {
+        }
+        
+        [Fact(DisplayName = "Linq2Db should recover data created using other persistence plugins")]
+        public async Task RecoveryValidationTest()
+        {
+            await ValidateRecovery(TestCluster!.ShardRegions[0]);
+        }
+
+        [Fact(DisplayName = "Linq2Db should be able to retrieve persistence ids on data created by other persistence plugin")]
+        public async Task PersistenceIdQueryValidationTest()
+        {
+            var system = TestCluster!.ActorSystems[0];
+            
+            var readJournal = PersistenceQuery.Get(system)
+                .ReadJournalFor<Linq2DbReadJournal>(Linq2DbReadJournal.Identifier);
+
+            var result = await readJournal.CurrentPersistenceIds().Where(id => !id.StartsWith("/"))
+                .RunAsAsyncEnumerable(system.Materializer()).ToListAsync();
+
+            result.Should().BeEquivalentTo(Enumerable.Range(0, 100).Select(i => i.ToString()));
+        }
+
+        [Fact(DisplayName = "Linq2Db should be able to query events by persistence id on data created by other persistence plugin")]
+        public async Task ByPersistenceIdQueryValidationTest()
+        {
+            var system = TestCluster!.ActorSystems[0];
+            var readJournal = PersistenceQuery.Get(system)
+                .ReadJournalFor<Linq2DbReadJournal>(Linq2DbReadJournal.Identifier);
+
+            foreach (var id in Enumerable.Range(0, 100))
+            {
+                var baseValue = id * 3;
+                var roundTotal = (baseValue * 3 + 3) * 4;
+
+                var events = await readJournal.CurrentEventsByPersistenceId(id.ToString(), 0, long.MaxValue)
+                    .RunAsAsyncEnumerable(system.Materializer()).ToListAsync();
```

Comment:
Happy that `RunAsAsyncEnumerable` is useful, although It does still scare me sometimes 😅

Reply:
Its pretty safe, as long as we know that the event source is not an infinite source